facebook/OpenBIC
# fby3.5-bb: Support sled/slot button SEL and slot button behavior

Summary:
- Support press and release sled/slot button sel.
- Support slot button behavior.
  - BB BIC will send command to BB CPLD to perform 12V-cycle, if slot button is pressed over 4 seconds.

Test Plan:
- Build code: Pass
- Check addsel or not after pressing sled/slot button: Pass
- Check slot to perform 12V-cycle after pressing slot button over 4 seconds: Pass

Note:
Config D button behavior

button type   Behavior
----------------------------------------------------------------------------------
sled button   server performs sled-cycle after pressing sled button over 4 seconds
slot button   slot performs 12V-cycle after pressing slot button over 4 seconds

Log:
-Sled button sel
1    slot1   2018-03-09 04:35:42  ipmid   SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 04:35:42, Sensor: BUTTON_DETECT (0xE2) , Event Data: (01FFFF) BB_BUTTON_BMC_BIC_N_R Assertion
1    slot1   2018-03-09 04:35:46  ipmid   SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 04:35:46, Sensor: BUTTON_DETECT (0xE2) , Event Data: (01FFFF) BB_BUTTON_BMC_BIC_N_R Deassertion

-Slot1 button sel
1    slot1   2018-03-09 04:37:52  ipmid   SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 04:37:52, Sensor: BUTTON_DETECT (0xE2)  , Event Data: (02FFFF) AC_ON_OFF_BTN_SLOT1_N Assertion
1    slot1   2018-03-09 04:37:57  ipmid   SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 04:37:57, Sensor: BUTTON_DETECT (0xE2)  , Event Data: (02FFFF) AC_ON_OFF_BTN_SLOT1_N Deassertion

-Slot3 button sel
1    slot1   2018-03-09 21:32:39  ipmid   SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 21:32:39, Sensor: BUTTON_DETECT (0xE2)  , Event Data: (03FFFF) AC_ON_OFF_BTN_SLOT3_N Assertion
1    slot1   2018-03-09 21:32:44  ipmid   SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 21:32:44, Sensor: BUTTON_DETECT (0xE2)  , Event Data: (03FFFF) AC_ON_OFF_BTN_SLOT3_N Deassertion

-Slot button behavior log
1   slot1  2018-03-09 04:37:52  ipmid        SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 04:37:52, Sensor: BUTTON_DETECT (0xE2) , Event Data: (02FFFF) AC_ON_OFF_BTN_SLOT1_N Assertion
1   slot1  2018-03-09 04:37:57  ipmid        SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 04:37:57, Sensor: BUTTON_DETECT (0xE2) , Event Data: (02FFFF) AC_ON_OFF_BTN_SLOT1_N Deassertion
1   slot1  2018-03-09 04:35:19  power-util   SERVER_POWER_ON successful for FRU: 1
6   nic    2018-03-09 04:35:21  ncsid        FRU: 6 NIC AEN Supported: 0x7, AEN Enable Mask=0x7
6   nic    2018-03-09 04:35:21  ncsid        FRU: 6 PLDM type supported = 0x35
6   nic    2018-03-09 04:35:21  ncsid        FRU: 6 PLDM type 0 version = 1.0.0.0
6   nic    2018-03-09 04:35:21  ncsid        FRU: 6 PLDM type 2 version = 1.1.0.0
6   nic    2018-03-09 04:35:21  ncsid        FRU: 6 PLDM type 4 version = 1.0.0.0
6   nic    2018-03-09 04:35:21  ncsid        FRU: 6 PLDM type 5 version = 1.0.0.0
6   nic    2018-03-09 04:35:21  ncsid        FRU: 6 PLDM sensor monitoring enabled
0   all    2018-03-09 04:35:21  healthd      SLED Powered OFF at Fri Mar  9 04:35:20 2018
0   all    2018-03-09 04:35:21  healthd      SLED Powered ON at Fri Mar  9 04:34:50 2018
0   all    2018-03-09 04:35:21  fan_check    Fan slot 0 is dual fan, please install single fan in DVT system
0   all    2018-03-09 04:35:21  fan_check    Fan slot 1 is dual fan, please install single fan in DVT system
